docs/reference/quadratic-selfonn.md

```diff
@@ -0,0 +1,225 @@
+
+```python
+QuadraticSelfOnnLearner.infer(self, batch)
+```
+
+This method is used to classify signals. Can be used to infer a single utterance, or a batch of signals.
+
+Parameters:
+
+- **batch**: *Timeseries* or *List*[*Timeseries*]   
+  Either a `Timeseries` or a list of `Timeseries`.
+
+#### `QuadraticSelfOnnLearner.save`
+
+```python
+QuadraticSelfOnnLearner.save(self, path)
+```
+
+This method saves the model in the directory provided by `path`.
+
+Parameters:
+
+- **path**: *str*  
+  Path to the directory where the model should be saved.
+  Does not need to exist before the function call.
+
+#### `QuadraticSelfOnnLearner.load`
+
+```python
+QuadraticSelfOnnLearner.load(self, path)
+```
+
+This method loads the model from the directory provided by `path`.
+In practice the same path as provided to `save` beforehand.
+
+Parameters:
+
+- **path**: *str*  
+  Path to the model directory to be loaded.
+
+#### `QuadraticSelfOnnLearner.download_pretrained`
+
+```python
+QuadraticSelfOnnLearner.download_pretrained(self, path)
+```
+
+This method downloads a pretrained model from the OpenDR FTP server.
+The pretrained model was trained on 20 classes of the GSC dataset in alphabetical classification order.
+A new directory will be created to the directory specified by path called "QuadraticSelfOnn", where the model will be saved.
+
+Parameters:
+
+- **path**: *str, default="."*   
+  Path to the parent directory where the model should be downloaded.
+
+#### Examples
+
+* **Training example using randomized data in place of recorded samples.**
+```python
+import numpy as np
+import os
+
+from opendr.engine.datasets import DatasetIterator
+from opendr.perception.speech_recognition.quadraticselfonn.quadraticselfonn_learner import QuadraticSelfOnnLearner
+
+class RandomDataset(DatasetIterator):
+    def __init__(self):
+        super().__init__()
+        
+    def __len__(self):
+        return 64
+
+    def __getitem__(self, item):
+        return np.random.rand(16000), np.random.choice(10)
+
+learner = QuadraticSelfOnnLearner(output_classes_n=10, iters=10, expansion_order=3)
+training_dataset = RandomDataset()
+validation_dataset = RandomDataset()
+
+results = learner.fit(dataset=training_dataset, val_dataset=validation_dataset)
+# Print the validation accuracy of the last epoch and save the model to a file
+print(results[10]["validation_results"]["test_accuracy"])
+learner.save(os.path.join(".", "example", "directory", "path", "model"))
+ ```
+
+* **Load an existing model and infer a sample from an existing file.**
+```python
+import librosa
+import numpy as np
+import os
+
+from opendr.engine.data import Timeseries
+from opendr.perception.speech_recognition.quadraticselfonn.quadraticselfonn_learner import QuadraticSelfOnnLearner
+
+learner = QuadraticSelfOnnLearner(output_classes_n=10)
+learner.load(os.path.join(".", "example", "directory", "path", "model"))
+
+signal, sampling_rate = librosa.load("command.wav", sr=learner.sample_rate)

```

Comment:
Maybe it would be good to also explain where does this file come from.
Because running this snippet as-is it throws the errors:
```
RuntimeError: Error opening 'command.wav': System error.
...
FileNotFoundError: [Errno 2] No such file or directory: 'command.wav'
```

Reply:
I added a note for the reader to clarify the source of the file in the example